@nativescript/core 8.5.4+: ReferenceError androidApp is not defined

### Issue Description

After upgrading @nativescript/core from 8.5.3 to 8.5.4 (or latest 8.5.6), our Android app is crashing for Android 8 and below when calling the **connectivity startMonitoring** method.

```
//app.js

const applicationModule = require("@nativescript/core/application");
const connectivityModule = require("@nativescript/core/connectivity");

applicationModule.on(applicationModule.suspendEvent, function(args) {
  connectivityModule.stopMonitoring();
});


applicationModule.on(applicationModule.resumeEvent, function(args) {
  connectivityModule.startMonitoring(function(newConnectionType) {
    console.log(newConnectionType);
  });
});

applicationModule.run({ moduleName: 'app-root' });
```

### Reproduction

app for reproduction:
[nsandroidcrash.zip](https://github.com/NativeScript/NativeScript/files/11847660/nsandroidcrash.zip)

Just start the app on Android 8 or below, then you will see the crash.

### Relevant log output (if applicable)

```shell
System.err: java.lang.RuntimeException: Unable to resume activity {org.nativescript.nsandroidcrash/com.tns.NativeScriptActivity}: com.tns.NativeScriptException: Calling js method onPostResume failed
  System.err: ReferenceError: androidApp is not defined
  System.err:   at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3429)
  System.err:   at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3469)
  System.err:   at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2732)
  System.err:   at android.app.ActivityThread.-wrap12(ActivityThread.java)
  System.err:   at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1477)
  System.err:   at android.os.Handler.dispatchMessage(Handler.java:102)
  System.err:   at android.os.Looper.loop(Looper.java:154)
  System.err:   at android.app.ActivityThread.main(ActivityThread.java:6119)
  System.err:   at java.lang.reflect.Method.invoke(Native Method)
  System.err:   at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
  System.err:   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
  System.err: Caused by: com.tns.NativeScriptException: Calling js method onPostResume failed
```


### Environment

<!-- COPY START -->
```yaml
OS: macOS 13.4
CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
Shell: /bin/zsh
node: 16.17.0
npm: 8.19.3
nativescript: 8.5.3

# android
java: 11.0.11
ndk: Not Found
apis: Not Found
build_tools: Not Found
system_images: Not Found

# ios
xcode: 14.3.1/14E300c
cocoapods: 1.12.1
python: 2.7.18
python3: 3.9.6
ruby: 2.6.10
platforms: 
  - DriverKit 22.4
  - iOS 16.4
  - macOS 13.3
  - tvOS 16.4
  - watchOS 9.4
```

### Dependencies

```json
"dependencies": {
  "@nativescript/core": "^8.5.6"
},
"devDependencies": {
  "@nativescript/android": "8.5.0",
  "@nativescript/webpack": "^5.0.16"
}
```
<!-- COPY END -->


### Please accept these terms

- [X] I have searched the [existing issues](https://github.com/NativeScript/NativeScript/issues) as well as [StackOverflow](https://stackoverflow.com/questions/tagged/nativescript) and this has not been posted before
- [X] This is a bug report
- [X] I agree to follow this project's [Code of Conduct](https://github.com/NativeScript/NativeScript/blob/master/tools/notes/CONTRIBUTING.md#coc)

--- a/packages/core/connectivity/index.android.ts
+++ b/packages/core/connectivity/index.android.ts
@@ -109,8 +109,7 @@ function startMonitoringLegacy(connectionTypeChangedCallback) {
 		connectionTypeChangedCallback(newConnectionType);
 	};
 	const zoneCallback = zonedCallback(onReceiveCallback);
-	// @ts-ignore
-	androidApp.registerBroadcastReceiver(android.net.ConnectivityManager.CONNECTIVITY_ACTION, zoneCallback);
+	Application.android.registerBroadcastReceiver(android.net.ConnectivityManager.CONNECTIVITY_ACTION, zoneCallback);
 }
 
 let callback;

--- a/packages/core/global-types.d.ts
+++ b/packages/core/global-types.d.ts
@@ -103,7 +103,7 @@ declare module globalThis {
 	function _isModuleLoadedForUI(moduleName: string): boolean;
 
 	var onGlobalLayoutListener: any;
-	function zonedCallback(callback: Function): Function;
+	function zonedCallback<T = Function>(callback: T): T;
 	var Reflect: any;
 	function Deprecated(target: Object, key?: string | symbol, descriptor?: any): any;
 	function Experimental(target: Object, key?: string | symbol, descriptor?: any): any;
@@ -358,7 +358,7 @@ declare function fail(data: any): void;
  */
 // declare function clearInterval(id: number): void;
 
-declare function zonedCallback(callback: Function): Function;
+declare function zonedCallback<T = Function>(callback: T): T;
 
 /**
  * Create a Java long from a number

